packages/components/src/custom-select-control/index.js

```diff
@@ -112,71 +69,67 @@ export default function CustomSelectControl( {
 			) }
 		>
 			{ hideLabelFromVision ? (
-				<VisuallyHidden as="label" { ...getLabelProps() }>
-					{ label }
-				</VisuallyHidden>
+				<VisuallyHidden as="label">{ label }</VisuallyHidden>
 			) : (
 				/* eslint-disable-next-line jsx-a11y/label-has-associated-control, jsx-a11y/label-has-for */
 				<label
-					{ ...getLabelProps( {
+					{ ...{
 						className: 'components-custom-select-control__label',
-					} ) }
+					} }
 				>
 					{ label }
 				</label>
 			) }
-			<Button
-				{ ...getToggleButtonProps( {
-					// This is needed because some speech recognition software don't support `aria-labelledby`.
-					'aria-label': label,
-					'aria-labelledby': undefined,
-					className: 'components-custom-select-control__button',
-					isSmall: true,
-					describedBy: getDescribedBy(),
-				} ) }
+			<DropdownMenu
+				className={ classnames(
+					'components-custom-select-control__inner',
+					className
+				) }
+				popoverProps={ POPOVER_PROPS }
+				toggleProps={ toggleProps }
+				icon={ null }
+				noIcons={ true }
 			>
-				{ itemToString( selectedItem ) }
-				<Icon
-					icon={ chevronDown }
-					className="components-custom-select-control__button-icon"
-				/>
-			</Button>
-			<ul { ...menuProps }>
-				{ isOpen &&
-					items.map( ( item, index ) => (
-						// eslint-disable-next-line react/jsx-key
-						<li
-							{ ...getItemProps( {
-								item,
-								index,
-								key: item.key,
-								className: classnames(
-									item.className,
-									'components-custom-select-control__item',
-									{
-										'is-highlighted':
-											index === highlightedIndex,
-										'has-hint': !! item.__experimentalHint,
-									}
-								),
-								style: item.style,
-							} ) }
-						>
-							{ item.name }
-							{ item.__experimentalHint && (
-								<span className="components-custom-select-control__item-hint">
-									{ item.__experimentalHint }
-								</span>
-							) }
-							{ item === selectedItem && (
-								<Icon
-									icon={ check }
-									className="components-custom-select-control__item-icon"
-								/>
-							) }
-						</li>
-					) ) }
-			</ul>
+				{ ( { isOpen, onClose } ) =>
+					isOpen && (
+						<MenuGroup>
+							{ items.map( ( item, index ) => (
+								<MenuItem
+									key={ index }
+									{ ...{
+										className: classnames(
+											item.className,
+											'components-custom-select-control__item',
+											{
+												'is-highlighted':
+													item === selectedItem,
+												'has-hint': !! item.__experimentalHint,
+												'has-check':
+													item === selectedItem,
+											}
+										),
+										style: item.style,
+									} }
+									onClick={ () => {
+										onSelectedItemChange( {
```

Comment:
The readme for this component [states the `onChange` prop is optional](https://github.com/WordPress/gutenberg/blob/ae56763775c178aa4e020551c445fe607bd49931/packages/components/src/custom-select-control/README.md#onchange) so `onSelectedItemChange` here might not be a function. We hit this error when testing the recent changes in the Storybook as well.